independent-projects/qute/core/src/main/java/io/quarkus/qute/JsonEscaper.java

```diff
@@ -1,10 +1,114 @@
 package io.quarkus.qute;
 
+import java.util.Arrays;
 import java.util.Optional;
 
 import io.quarkus.qute.TemplateNode.Origin;
 
-public class JsonEscaper extends CharReplacementResultMapper {
+public final class JsonEscaper implements ResultMapper {
```

Comment:
I like the idea and escpecially the numbers :smile: but would it be possible to extract the logic in a subclass so that we could use it for HtmlEscaper as well?

Reply:
Doing it can cause a dangerous thing which kill performance i.e. a method called "replacementOf" which is implemented differently in two subclasses.
If we end up doing it and the runtime app use both the subclass this will cause a massive slowdown in both since some optimizations around inlining and loop unrolling will be lost.
Sadly the most powerful optimization in these tight loops is.."copying code" (I know sound weird but I can explain more in a call if you wish 🙏)

Reply:
But if I can avoid having special handling without subclassing, that would work i.e. just different tables and fields to drive decisions...
Let's see what I can do 🙏 
But I need to be clear what I have done to make it fast; if they part isn't clear, let's speak so I can help ❤️